source/server/configuration_impl.cc

```diff
@@ -30,8 +30,14 @@ namespace Server {
 namespace Configuration {
 
 bool FilterChainUtility::buildFilterChain(Network::FilterManager& filter_manager,
-                                          const std::vector<Network::FilterFactoryCb>& factories) {
-  for (const Network::FilterFactoryCb& factory : factories) {
+                                          const Filter::NetworkFilterFactoriesList& factories) {
+  for (const auto& filter_config_provider : factories) {
+    auto config = filter_config_provider->config();
+    if (!config.has_value()) {
+      return false;
+    }
+
+    auto factory = config.value();
```

Comment:
Same here - let's make sure these are taken as references, not value copy.

Reply:
done